application/controllers/QuestionAdministrationController.php

```diff
@@ -1730,6 +1730,42 @@ public function actionGetAdvancedSettingsHTML(int $surveyId, string $questionTyp
         );
     }
 
+    /**
+     * Get HTML for extra options (subquestions/answers).
+     * Called with Ajax after question type is selected.
+     *
+     * @param int $surveyId
+     * @param string $questionType One-char string
+     * @param int $questionId Null or 0 if new question is being created.
+     * @return void
+     */
+    public function actionGetExtraOptionsHTML(int $surveyId, string $questionType, $questionId = null)
+    {
+        if (empty($questionType)) {
```

Comment:
I _think_ this check might be superfluous? Since Yii checks the arguments to the action in the request handling.

Reply:
You are saying Yii check's for empty question types?

This was taken from actionGetExtraOptionsHTML.
This was supposed to be a quick win stuff, do you want to review this and update it now?

Reply:
Quick win is illegal if it adds technical debt. ;)

Reply:
> You are saying Yii check's for empty question types?

I'm saying Yii is checking that the arguments are correct to the controller action. Try it.

Reply:
Since the type is not nullable, I mean.

Reply:
Oh, but empty checks for empty string too, perhaps... Gah.

Reply:
I would just leave it like this.

> Quick win is illegal if it adds technical debt. ;)

I don't think we are doing that, but won't go into that discussion :) I understand what you mean.

Reply:
![image](https://user-images.githubusercontent.com/4942495/120323262-b4a02880-c2bb-11eb-8442-75e692d7c2e7.png)
This is like when calling it with empty questionType (like this http://localhost:8000/limesurvey/index.php/questionAdministration/getExtraOptionsHTML?surveyId=586669&questionId=5700&YII_CSRF_TOKEN=MGxBVTZxaF9tZ3FNdm5fTmJZSXhmQ3BnRnE2eTQyaGS_eBGxzqzMo3r17jyEXq_KS19jJTj37hscsov6Tu9Qbg%3D%3D&questionType=)

Reply:
What happen if questionType is a string but no related questionType ? 

We know it's a string (can be empty) but we need to check if it's not a know questiontype (and if not : it's a 404 ?)

Reply:
Must fix diffarently : /questionAdministration/getExtraOptionsHTML?surveyId=753953&questionId=7865&questionType=AA

![Capture d’écran_2021-06-01_14-37-24](https://user-images.githubusercontent.com/1439428/120324475-f6929400-c2e6-11eb-9691-d97cffc05c01.png)

empty page with debug = 0

questionAdministration/getGeneralSettingsHTML?surveyId=753953&questionId=7865&&questionType=EE&questionTheme=bootstrap_buttons
![Capture d’écran_2021-06-01_14-39-04](https://user-images.githubusercontent.com/1439428/120324639-2346ab80-c2e7-11eb-84de-99159306f0db.png)

Reply:
Stop hacking the URL, Denis. ><

Reply:
> So I guess we need it

Yep, I was wrong.

Reply:
We need more, not only empty :) 

> Stop hacking the URL, Denis. ><

1. Never trust user
2. JS always broke